pydantic_settings/sources/base.py

```diff
@@ -227,7 +227,7 @@ def __init__(self, settings_cls: type[BaseSettings], nested_model_default_partia
             else self.config.get('nested_model_default_partial_update', False)
         )
         if self.nested_model_default_partial_update:
-            for field_name, field_info in settings_cls.model_fields.items():
+            for field_name, field_info in settings_cls.__pydantic_fields__.items():
```

Comment:
```suggestion
            for field_name, field_info in settings_cls.model_fields.items():
```

and others where it is accessed on a class